test/gtest/log_test_helper.hpp

```diff
@@ -0,0 +1,196 @@
+#if MIOPEN_BACKEND_OPENCL
+        clReleaseMemObject(data);
+#elif MIOPEN_BACKEND_HIP
+        hipFree(data);
+#endif
+    }
+};
+
+struct Conv
+{
+    Tensor input;
+    Tensor output;
+    Tensor weights;
+    miopenConvolutionDescriptor_t convDesc;
+
+    Conv() : input(128, 3, 32, 32), output(128, 64, 32, 32), weights(64, 3, 3, 3)
+    {
+        EXPECT_EQ(miopenCreateConvolutionDescriptor(&convDesc), 0);
+        EXPECT_EQ(miopenInitConvolutionDescriptor(convDesc, miopenConvolution, 1, 1, 1, 1, 1, 1),
+                  0);
+    }
+    ~Conv() { miopenDestroyConvolutionDescriptor(convDesc); }
+};
+
+bool isSubStr(const std::string& str, const std::string& sub_str)
+{
+    return str.find(sub_str) != std::string::npos;
+}
+
+#if MIOPEN_BACKEND_OPENCL
+#define BKEND "OpenCL"
+#elif MIOPEN_BACKEND_HIP
+#define BKEND "HIP"
+#endif
+static const std::string& logConv =
+    "MIOpen(" BKEND
+    "): Command [LogCmdConvolution] ./bin/MIOpenDriver conv -n 128 -c 3 -H 32 -W 32 -k "
+    "64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -F 1 -t 1";
+static const std::string& logFindConv =
+    "MIOpen(" BKEND
+    "): Command [LogCmdFindConvolution] ./bin/MIOpenDriver conv -n 128 -c 3 -H 32 -W 32 "
+    "-k 64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -F 1 -t 1";
+
+static const std::string& envConv     = "MIOPEN_ENABLE_LOGGING_CMD";
+static const std::string& envFindConv = "MIOPEN_ENABLE_LOGGING_CMD_FIND";
+
+// We capture the std::cerr that was generated from the log functions
+// into a string and use that string for assertion.
+inline void TestLogFun(std::function<void(const miopenTensorDescriptor_t&,
+                                          const miopenTensorDescriptor_t&,
+                                          const miopenConvolutionDescriptor_t&,
+                                          const miopenTensorDescriptor_t&,
+                                          const miopen::ConvDirection&,
+                                          bool)> const& func,
+                       std::string env_var,
+                       std::string sub_str,
+                       bool set_env)
+{
+    // start capturing std::cerr
+    CerrRedirect capture_cerr;
+    // prepare tensor and convolution descriptors
+    Conv test_conv_log;
+    if(set_env)
+        Env::setEnvironmentVariable(env_var, "1");
+    else
+        Env::unSetEnvironmentVariable(env_var);
+
+    func(test_conv_log.input.desc,
+         test_conv_log.weights.desc,
+         test_conv_log.convDesc,
+         test_conv_log.output.desc,
+         miopen::ConvDirection::Fwd,
+         false);
+
+    // get the captured string
+    std::string str = capture_cerr.getString();
+    // now do the assertions
+    if(set_env)
+        ASSERT_TRUE(isSubStr(str, sub_str)) << "str     : " << str << "str_sub : " << sub_str;
+    else
+        ASSERT_FALSE(isSubStr(str, sub_str)) << "str     : " << str << "str_sub : " << sub_str;
+}
```

Comment:
Actually, AFAICS this file should contain only declarations of:
```
TestLogFun
envConv
envFindConv
logConv
logFindConv
```
The definitions of all entities shall be be moved to log_test_helper.cpp.

Reply:
modified code as per request.